sys/arduino/base.cpp

```diff
@@ -58,3 +61,44 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+#ifdef ADC_NUMOF
+/*
+ * Bitfield for the state of the ADC-channels.
+ * 0: Not initialized
+ * 1: Successfully initialized
+ */
+ static uint16_t adc_state;
+
+int analogRead(int arduino_pin)
```

Comment:
to make sure we are on the same page: arduino ADC pins (the value of `arduion_pin`) are handled as unsigned integers starting from 0 for ARDUINO_A0 to x. To quote from [here](https://www.arduino.cc/en/Reference/AnalogRead):
```
Syntax

analogRead(pin)

Parameters

pin: the number of the analog input pin to read from (0 to 5 on most boards, 0 to 7 on the Mini and Nano, 0 to 15 on the Mega)
```